udata/features/territories/views.py

```diff
@@ -72,50 +73,74 @@ def compute_territory_dataset(territory, dataset, resource_id):
                      attachment_filename=attachment_filename)
 
 
-@blueprint.route('/territory/regions/', endpoint='home')
+@blueprint.route('/territoires/', endpoint='home')
 def render_home():
     if not current_app.config.get('ACTIVATE_TERRITORIES'):
         return abort(404)
 
-    regions = GeoZone.objects.get(id='country/fr').children
+    regions = GeoZone.objects(level='fr/region').valid()
+    regions = sorted(
+        regions,
+        key=lambda zone: unicodedata.normalize('NFD', zone.name)
+                                    .encode('ascii', 'ignore'))
 
     return theme.render('territories/home.html', **{
         'geojson': {
             'type': 'FeatureCollection',
             'features': [region.toGeoJSON() for region in regions]
-        }
+        },
+        'regions': regions
     })
 
 
 @blueprint.route('/town/<int:code>/', endpoint='town')
 def redirect_town(code):
     """
-    Legacy redirect now prefixed with `territories`.
+    Legacy redirect now prefixed with `territories` + French name.
     """
     # Turn the dict into a namedtuple to be consistent in routing.
     territory = dict_to_namedtuple('Territory',
-                                   {'code': code, 'level_name': 'town'})
+                                   {'code': code, 'level_name': 'commune'})
     return redirect(url_for('territories.territory', territory=territory))
 
 
-@blueprint.route('/territory/<territory:territory>/', endpoint='territory')
+@blueprint.route('/territoires/<territory:territory>/', endpoint='territory')
 def render_territory(territory):
     if not current_app.config.get('ACTIVATE_TERRITORIES'):
         return abort(404)
 
-    DATASETS = TERRITORY_DATASETS[territory.level_name]
+    # Check that this territory is still valid (redirect old French
+    # regions for instance).
+    if territory.validity.get('end'):
```

Comment:
TODO: that's where we want to use [GeoHisto](https://github.com/etalab/geohisto) data.